docs/howto/pipeline_testing.md

```diff
@@ -0,0 +1,183 @@
+# HOWTO: Writing pipeline tests for a package
+
+## Introduction
+
+Elastic Packages are comprised of data streams. A pipeline test exercises ingest pipelines defined for a package's data stream.
```

Comment:
Nit: should we say "Elasticsearch Ingest Node pipelines" instead of "ingest pipelines" to make it completely unambigious?

Reply:
Fixed